ir/src/symbol_table.rs

```diff
@@ -65,7 +65,10 @@ impl SymbolTable {
     /// Sets the number of trace segments to the maximum of `num_trace_segments` and the provided
     /// trace segment identifier, which is indexed from zero.
     fn set_num_trace_segments(&mut self, trace_segment: TraceSegment) {
-        self.num_trace_segments = self.num_trace_segments.max((trace_segment + 1).into())
+        // self.num_trace_segments = self.num_trace_segments.max((trace_segment + 1).into())
+        if trace_segment >= self.segment_width.len() as u8 {
+            self.segment_width.resize(trace_segment as usize + 1, 0);
+        }
```

Comment:
The method should be renamed if the functionality is changing. The comment should be similarly updated. Also, the commented-out code can be deleted.

It's not clear to me what's happening here. Once these things are updated, I may have other thoughts.

Reply:
The idea was to combine information about number of segments and number of columns in each segment. Since now we have a `segment_widths` vector, the number of segments can be obtained as length of this vector. And if we want to update the segments number we should resize `segment_widths` vector (but only if new segment is greater than current biggest one). I forgot to update the comment, but I didn't change the name of the function intentionally so that the operation with `SymbolTable` would remain the same, because the function continued to return the exact same value. But I can change that if you think it's better to update the name.

Reply:
Yes, please update the name. I don't think the name reflects the functionality anymore.